src/import/FSHImporter.ts

```diff
@@ -1916,6 +1961,67 @@ export class FSHImporter extends FSHVisitor {
     }
   }
 
+  /**
+   * Given a path and the context containing it, apply the path context indicated by the path's indent
+   * @param path - The path to apply context to
+   * @param parentCtx - The parent element containing the path
+   * @param isPathRule - Flag to indicate that the path is for a PathRule which is only used to set context
+   * @returns {string} - The path with context prepended
+   */
+  private prependPathContext(path: string, parentCtx: any, isPathRule: boolean): string {
+    const location = this.extractStartStop(parentCtx);
+    const currentIndent = location.startColumn - this.baseIndent;
+
+    if (currentIndent > 0 && this.pathContext.length === 0) {
+      logger.error(
+        'The first rule of a definition cannot be indented. The rule will be processed as if it is not indented.',
+        location
+      );
+      return path;
+    }
+
+    if (currentIndent % INDENT_WIDTH !== 0 || currentIndent < 0) {
+      logger.error(
+        `Unable to determine path context for rule indented ${currentIndent} space(s). Rules must be indented in multiples of ${INDENT_WIDTH} space(s).`,
+        location
+      );
+      return path;
+    }
+
+    // And we require that rules are not indented too deeply
+    const contextIndex = currentIndent / INDENT_WIDTH;
+    if (contextIndex > this.pathContext.length) {
+      logger.error(
+        `Cannot determine path context of rule since it is indented too deeply. Rules must be indented in increments of ${INDENT_WIDTH} space(s).`,
+        location
+      );
+      return path;
+    }
+
+    // If the element is not indented, just reset the context
+    // Replace '[+]' with '[=]' in the version of the path to add to pathContext, unless the rule is only used to set a path
+    const newContext = isPathRule ? path : path.replace(/\[\+\]/g, '[=]');
+    if (contextIndex === 0) {
+      this.pathContext = [newContext];
+      return path;
+    }
+
+    const currentContext = this.pathContext[contextIndex - 1];
+    if (currentContext === '') {
+      logger.error(
+        'Rule cannot be indented below rule which has no path. The rule will be processed as if it is not indented.',
+        location
+      );
+      return path;
+    }
+
+    this.pathContext.splice(contextIndex);
+    // Once we have used the existing context, clear it of any [+] so that a rule that is only setting path only applies [+] once
+    this.pathContext = this.pathContext.map(pc => pc.replace(/\[\+\]/g, '[=]'));
```

Comment:
Is it necessary to do this on every element of the array?  I just can't think of how anything but the last element could every have a `[+]` (because I think all elements before the last element have been through this code before?).   But I may not be understanding it 100% right.

Reply:
No I think you're right, I don't think it could ever be on an element that isn't the last element.

Reply:
Fixed in https://github.com/FHIR/sushi/pull/797/commits/c949df9f60f390f48712f29cce70216e0e7c826d.